File: HRIS/Hris.Web.Main/ClientApp/app/components/shared/datamodel/response.result.ts
import { ResultCode } from "../enum";

export class ResponseResult {
    constructor(code: ResultCode, data: any, message: string) {
        this.code = code;
        this.data = data;
        this.message = message;
    }
    code: ResultCode;
    data: any;
    message: string;
}

File: new_hris/Hris/Hris.Shared/ResponseResult.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Hris.Shared
{
    public class ResponseResult<T>
    {
        public ResponseResult(T data)
        {
            Data = data;
            Code = ResultCode.Success;
        }

        public ResponseResult(string message)
        {
            Message = message;
            Code = ResultCode.Error;
        }

        public ResponseResult(T data, ResultCode code, string message)
        {
            Data = data;
            Code = code;
            Message = message;
        }

        public ResultCode Code { get; set; }
        public T Data { get; set; }
        public string Message { get; set; }
    }
}
